metrics/first-meaningful-paint/gather.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var FirstMeaningfulPaint = {
+  run: function(driver, url) {
+    return driver.disableCaching()
+      // Fire up the trace.
+      .then(driver.beginTrace)
+
+      // Go to the URL.
+      .then(_ => driver.gotoURL(url, driver.WAIT_FOR_LOAD))
+
+      // Stop the trace, which captures the records.
+      .then(driver.endTrace)
+      .then(traceContents => ({traceContents}));
```

Comment:
this work and the work done here -> https://github.com/GoogleChrome/lighthouse/pull/29/files#diff-f0e1982b7f0bb53cbfb9ad52ce2d34f6R45

look like they might be able to be condensed :fountain: